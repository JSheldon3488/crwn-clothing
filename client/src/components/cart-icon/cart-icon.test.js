import React from 'react';
import { shallow } from 'enzyme';
import CartIcon from './cart-icon.component';

describe('CartIcon Component Tests', () => {
  let wrapper;
  let mockToggleCartHidden;
  
  beforeEach(() => {
    mockToggleCartHidden = jest.fn();

    const mockProps = {
      itemCount: 5,
      toggleCartHidden: mockToggleCartHidden
    };

    wrapper = shallow(<CartIcon {...mockProps} />);
  });

  it('should render and match CartIcon snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call toggleCartHidden when icon is clicked', () => {
    wrapper.find('CartIconStyles').simulate('click');
    expect(mockToggleCartHidden).toHaveBeenCalled();
  });

  it('should render the itemCount as the text', () => {
    const itemCount = parseInt(wrapper.find('ItemCountStyles').text());
    expect(itemCount).toBe(5);
  });
});