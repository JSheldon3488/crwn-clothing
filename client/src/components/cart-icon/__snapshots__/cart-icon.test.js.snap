// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartIcon Component Tests should render and match CartIcon snapshot 1`] = `ShallowWrapper {}`;
